### ISSUE TITLE
[naming-convention] Handle constructor passed as parameter

### ISSUE THREAD

**Issue by sindresorhus** [typescript-eslint/typescript-eslint#1484]
```js
{
	rules: {
		'@typescript-eslint/naming-convention': [
			'error',
			{
				selector: 'default',
				format: [
					'strictCamelCase'
				],
				leadingUnderscore: 'allow',
				trailingUnderscore: 'allow'
			}
		}
	}
}
```

```ts
const createAgentSpy = <T extends HttpAgent>(AgentClass: Constructor) => {
	const agent: T = new AgentClass({keepAlive: true});
	// ...
};
```

**Expected Result**

In some places I'm passing a class constructor as a parameter and I need to keep it PascalCase so I can `new` it without it looking weird and other lint rules triggering on the `new` call. I could rename it after passing it, but I don't think I should change my valid code to appease this rule. I'm not sure how we should solve this, but would be nice if there was a way to handle this.

**Actual Result**

```
✖   34:46  Parameter name AgentClass must match one of the following formats: strictCamelCase                      @typescript-eslint/naming-convention
```

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.17.0` |
| `@typescript-eslint/parser`        | `2.17.0` |
| `TypeScript`                       | `3.7.5` |
| `ESLint`                           | `6.8.0` |
| `node`                             | `10.17.0` |
| `npm`                              | `6.13.6` |

**Comment by bradzacher** [comment]
As with #1485, happy to have a new `type` option added for `constructable` or something.
Would be almost the same as the `function` type, but using `t.getConstructSignatures()` instead.

https://github.com/typescript-eslint/typescript-eslint/blob/fcff5b430c669b7ec27705aa18211e0eedfb486a/packages/eslint-plugin/src/rules/naming-convention.ts#L1256

Note that this would require type information to solve, as there's no other way to tell.

**Comment by JoshuaKGoldberg** [comment]
Per #8792, we're feature freezing this rule. It's reached the point of complexity where we're having a hard time justifying working on it. 

If you'd like to see this option happen, I'd encourage you to add it to a community project such as [eslint.style](https://eslint.style). Cheers! 💙
